Jenkinsfile

```diff
@@ -206,6 +217,17 @@ pipeline {
         }
     }
     post {
+        success {
+            // Docker images should remain upon failure for troubleshooting purposes.  However,
+            // if the stage is successful, there is no reason to look back at the Docker image.  In theory
+            // broken builds will eventually be fixed so this step should run eventually for every PR
+            sh(
+                label: 'Delete Docker Image on Success',
+                script: '''
+                    docker image prune --all --force --filter label=build=${BUILD_TAG};
+                '''
+            )
+        }
```

Comment:
A question, are the failed images not purged by the prune command in cleanup? That would be my assumption but they may be considered in use?

Reply:
That's a good question, TBH I copied this over from another Jenkinsfile. I don't even know that we want to keep the Docker image around to inspect on failure. We should get sufficient data from the logs and Allure report. 

If there was some situation where we were getting a repeat issue that we couldn't figure out and couldn't be reproduced anywhere else, then we could replay the job without the cleanup.

Any opposition to deleting all Docker images created by the job?
Ping @chrisveilleux

Reply:
From the docker documentation:
Remove all dangling images. If -a is specified, will also remove all images not referenced by any container.

The prune without --all, which is how it was before this PR just cleans up the dangling images.  Using --all cleans up all images not referenced by a container, which is too aggressive, so we use --all and --filter to delete the images related to the PR will be removed.